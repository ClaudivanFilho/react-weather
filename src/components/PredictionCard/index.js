import React, { Component, PropTypes } from 'react';

import Moment from 'react-moment';
import './index.scss';

export default class PredictionCard extends Component {
  
  render() {
    const {date, icon, temp, info, imageWidth} = this.props;
    return (
      <div className={`prediction-card`}>
        <h5>
          <Moment unix format="DD/MMM HH:mm">
            {date}
          </Moment>
        </h5>
        <div className="div-image col-sm-12">
          <span style={{color: "red", opacity: 1, marginTop:'15px', fontWeight: 'bold'}}>
            {temp && temp.toFixed(0)}Â°C 
          </span> 
          { icon && <img src={`//openweathermap.org/img/w/${icon}.png`} width={imageWidth || 50}/> }
        </div>
        <div className="col-sm-12">
          <h5>
             {info}
          </h5>
        </div>
      </div>
    );
  }
}

