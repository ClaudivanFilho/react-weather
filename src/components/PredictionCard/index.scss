.prediction-card {
  text-align: center;

  background: rgb(178,225,255); /* Old browsers */
  background: -moz-linear-gradient(top, rgba(178,225,255,1) 0%, rgba(102,182,252,1) 100%); /* FF3.6-15 */
  background: -webkit-linear-gradient(top, rgba(178,225,255,1) 0%,rgba(102,182,252,1) 100%); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(to bottom, rgba(178,225,255,1) 0%,rgba(102,182,252,1) 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#b2e1ff', endColorstr='#66b6fc',GradientType=0 );

  -webkit-box-shadow: 0px 0px 12px -2px rgba(0,0,0,0.71);
  -moz-box-shadow: 0px 0px 12px -2px rgba(0,0,0,0.71);
  box-shadow: 0px 0px 12px -2px rgba(0,0,0,0.71);
  margin: 5px;
  margin-left: 10px;
  margin-right: 10px;
  padding: 10px;
  border-radius: 10px;
}
