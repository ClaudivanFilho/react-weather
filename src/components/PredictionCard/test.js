import React from 'react';
import {shallow} from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';

import PredictionCard from './index';

configure({ adapter: new Adapter() });

const predictionMock = {
  date: '131414020',
  icon: '10n',
  info: 'Clouds'
}

test('Test PredictionCard Component - with icon', () => {
  const component = shallow(<PredictionCard {...predictionMock} />);
  expect(component.find('img').length).toEqual(1);
  expect(component.find('h5').length).toEqual(2);
});

test('Test PredictionCard Component - without icon', () => {
  const component = shallow(<PredictionCard />);
  expect(component.find('img').length).toEqual(0);
  expect(component.find('h5').length).toEqual(2);
});

test('Test PredictionCard Snapshot', () => {
  const component = renderer.create(
    <PredictionCard {...predictionMock}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})