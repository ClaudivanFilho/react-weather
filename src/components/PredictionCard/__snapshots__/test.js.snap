// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PredictionCard Snapshot 1`] = `
<div
  className="prediction-card"
>
  <h5>
    <time
      dateTime={"1974-03-01T23:53:40.000Z"}
    >
      01/Mar 20:53
    </time>
  </h5>
  <div
    className="div-image col-sm-12"
  >
    <span
      style={
        Object {
          "color": "red",
          "fontWeight": "bold",
          "marginTop": "15px",
          "opacity": 1,
        }
      }
    >
      Â°C
    </span>
    <img
      src="//openweathermap.org/img/w/10n.png"
      width={50}
    />
  </div>
  <div
    className="col-sm-12"
  >
    <h5>
      Clouds
    </h5>
  </div>
</div>
`;
